src/renderer/components/outputs/LargeImageOutput.tsx

```diff
@@ -10,12 +10,22 @@ import { GlobalContext, GlobalVolatileContext } from '../../contexts/GlobalNodeS
 import { OutputProps } from './props';
 
 interface LargeImageBroadcastData {
-    image: string;
+    '256': string;
+    '512': string;
+    '1024': string;
+    '2048': string;
     width: number;
     height: number;
     channels: number;
 }
 
+const pickImage = (last: LargeImageBroadcastData, zoom: number) => {
+    if (zoom < 1) return last['256'];
```

Comment:
This method of picking makes various assumptions:
1. It assumes no DPI scaling/zooming via Electron.
2. It assumes that the preview is (roughly) 200px (CSS units).

What we are really interested in to pick the right preview is this question: How many *screen* pixels is the preview container currently?

So we want `realSize = cssSize * zoom * dpiScaling * electronZoom`. In our case, `cssSize` is the 200px of the preview container. I think the browser already has a method to give you `dpiScaling * electronZoom`, so you can use that. 

It's quite easy to pick the right preview image with `realSize` and it also makes it easy to pick the right interpolation method (`imageRendering: pixelated/auto`).